app/Jobs/GenerateHashValue.php

```diff
@@ -114,30 +119,33 @@ protected function getThumbnail(Image $image)
     protected function getHashValue(Image $image)
     {
         $imageByteString = $this->getThumbnail($image);
-        dd($imageByteString);
-
-        $script = config('biigle.hash_value_generator');
-
-        try {
-            $inputPath = $this->createInputJson($image, $imageByteString);
-            $outputPath = $this->getOutputJsonPath($image);
-            $output = $this->python("{$script} {$inputPath} {$outputPath}");
-            $hashValue = json_decode(File::get($outputPath), true);
-        } catch (Exception $e) {
-            $input = File::get($inputPath);
-            throw new Exception("Input: {$input}\n" . $e->getMessage());
-        } finally {
-            if (isset($inputPath)) {
-                $this->maybeDeleteFile($inputPath);
-            }
-
-            if (isset($outputPath)) {
-                $this->maybeDeleteFile($outputPath);
+        if ($imageByteString !== false) {
+            $script = config('biigle.hash_value_generator');
+
+            try {
+                $outputPath = $this->getOutputJsonPath($image);
+                $inputPath = $this->createInputJson($image, $imageByteString);
+                $output = $this->python("{$script} {$inputPath} {$outputPath}");
+                $hashValue = decodeOutputJson($outputPath);
```

Comment:
Why is this function undefined? 
I just put this code part in a seperated function to overlay the funtion while testing. But the code doesn't find this function (Line 226)

Reply:
You have to use `$this`:
```suggestion
                $hashValue = $this->decodeOutputJson($outputPath);
```